test/suites/drift/suite_test.go

```diff
@@ -335,5 +337,71 @@ var _ = Describe("Drift", Label("AWS"), func() {
+
+			updatedAWSNodeTemplate := awstest.AWSNodeTemplate(*nodeTemplate.Spec.DeepCopy(), nodeTemplateSpec)
+			updatedAWSNodeTemplate.ObjectMeta = nodeTemplate.ObjectMeta
+			updatedAWSNodeTemplate.Annotations = map[string]string{v1alpha1.AnnotationNodeTemplateHash: updatedAWSNodeTemplate.Hash()}
+
+			env.ExpectCreated(pod, nodeTemplate, provisioner)
+			machine := env.EventuallyExpectCreatedMachineCount("==", 1)[0]
+			node := env.EventuallyExpectCreatedNodeCount("==", 1)[0]
+			env.EventuallyExpectHealthy(pod)
+
+			env.ExpectCreatedOrUpdated(updatedAWSNodeTemplate)
+
+			By("validating the drifted status condition has propagated")
+			EventuallyWithOffset(1, func(g Gomega) {
+				g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(machine), machine)).To(Succeed())
+				g.Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted)).ToNot(BeNil())
+				g.Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted).IsTrue()).To(BeTrue())
+			}).Should(Succeed())
+
+			delete(pod.Annotations, v1alpha5.DoNotEvictPodAnnotationKey)
+			env.ExpectUpdated(pod)
+			env.EventuallyExpectNotFound(pod, node)
+		},
+			Entry("InstanceProfile Drift", "InstanceProfile", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{InstanceProfile: awssdk.String("instance-profile-drift")}}),
+			Entry("UserData Drift", "UserData", v1alpha1.AWSNodeTemplateSpec{UserData: awssdk.String("#!/bin/bash\n/etc/eks/bootstrap.sh")}),
+			Entry("Tags Drift", "Tags", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{Tags: map[string]string{"keyTag-test-3": "valueTag-test-3"}}}),
+			Entry("MetadataOptions Drift", "MetadataOptions", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{LaunchTemplate: v1alpha1.LaunchTemplate{MetadataOptions: &v1alpha1.MetadataOptions{HTTPPutResponseHopLimit: awssdk.Int64(1)}}}}),
+			Entry("BlockDeviceMappings Drift", "BlockDeviceMappings", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{LaunchTemplate: v1alpha1.LaunchTemplate{BlockDeviceMappings: []*v1alpha1.BlockDeviceMapping{
+				{
+					DeviceName: awssdk.String("/dev/xvda"),
+					EBS: &v1alpha1.BlockDevice{
+						VolumeSize: resource.NewScaledQuantity(20, resource.Giga),
+						VolumeType: awssdk.String("gp3"),
+						Encrypted:  awssdk.Bool(true),
+					},
+				}}}}}),
+			Entry("DetailedMonitoring Drift", "DetailedMonitoring", v1alpha1.AWSNodeTemplateSpec{DetailedMonitoring: awssdk.Bool(true)}),
+			Entry("AMIFamily Drift", "AMIFamily", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{AMIFamily: awssdk.String(v1alpha1.AMIFamilyBottlerocket)}}),
+		)
 	})
 })
+
+func ExpectInstanceProfileCreated(ants v1alpha1.AWSNodeTemplateSpec) {
+	By("creating an instance profile")
+	ants.AWS.InstanceProfile = awssdk.String(*ants.AWS.InstanceProfile + "-" + settings.FromContext(env.Context).ClusterName)
+	createInstanceProfile := &iam.CreateInstanceProfileInput{
+		InstanceProfileName: ants.AWS.InstanceProfile,
+		Tags: []*iam.Tag{
+			{
+				Key:   awssdk.String("testing.karpenter.sh/discovery"),
```

Comment:
I believe this key should be specified from a constant value (I forget what the name of that variable is)